### ISSUE TITLE
Fix S2930 FP: Do not report on fields of IDisposable classes

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#8336]
```
public class Sample : IDisposable
{
    private readonly CancellationTokenSource source;

    public Sample() =>
        source = new(); // FP

   public void Dispose() =>
        source.Dispose();
}
```

and also
```
public class Sample : IDisposable
{
    private readonly CancellationTokenSource source;

    public Sample() =>
        source = new(); // FP, the fact that disposable field was not disposed should be another rule

   public void Dispose()
   {
        // Did we forget to dispose something here?
   }
}
```

**Comment by Tim-Pohlmann** [comment]
Multiple things:
1. I removed the CFG/SE label because this rule is not Symbolic Exection-based, nor does it use the CFG. That being said, we could consider rewriting it via SE.
2. The first sample does not reproduce (anymore?).
3. Regarding the second sample, I want to challenge the idea of having a separate rule for this case. S2930 is intelligent enough to differentiate between samples #1 and #2. Therefore, I find it unnecessary to add more complexity to the implementation by splitting it into multiple rules.
4. Under the premise that the second sample should trigger a different rule: The other rule does not exist yet. Thus, suppressing this case would cause more harm than good => I'm closing this issue.
